### ISSUE TITLE
nuxt/require-func-head doesnt work with factories

### ISSUE THREAD

**Issue by pimlie** [nuxt/eslint-plugin-nuxt#94]
This nuxt/require-func-head rule doesnt detect factories. Im using a head factory method which only returns a function when its needed.

So I need the code to be:
```js
head: createHead({})
```
but instead this rule transforms it into
```js
head() {
  return createHead({})
}
```
which wont work if createHead returns an fn itself (which is most cases)
